File: matrix_multiplication/matrix_mult.py
import timeit



setup = """
import numpy as np
A = np.random.rand(1000, 1000)
B = np.random.rand(1000, 1000)


def matrix_multiply(a, b):
    return a @ b


def hadamard_multiply(a, b):
    return a * b

"""


if __name__ == '__main__':
    print("Matrix multiplication:")
    num = 500
    time = timeit.timeit("matrix_multiply(A, B)", setup=setup, number=num)
    print(time / num, 's')
    print()
    print("Hadamard multiplication:")
    time = timeit.timeit("hadamard_multiply(A, B)", setup=setup, number=num)
    print(time / num, 's')

File: matrix_multiplication/matrix_mult.jl
using LinearAlgebra
using Random
using BenchmarkTools


function matrix_multiply(a, b)
    a * b
end


function time_multiply()
    print("Matrix multiplication:")
    @benchmark matrix_multiply(A, B) setup=(A=rand(Float64, (10000, 10000)); B=rand(Float64, (10000, 10000)))
end


time_multiply()
